libs/wire-api/src/Wire/API/Routes/Version.hs

```diff
@@ -63,8 +64,20 @@ data Version = V0 | V1 | V2 | V3
 
 instance ToSchema Version where
   schema =
-    enum @Integer "Version" . mconcat $
-      (\v -> element (fromIntegral $ fromEnum v) v) <$> [minBound @Version ..]
+    assert
+      (explicit == constructed)
+      (enum @Integer "Version" . mconcat . fmap (uncurry element) $ explicit)
+    where
+      constructed, explicit :: [(Integer, Version)]
+      constructed =
+        -- (fromEnum doesn't work as soon as we drop support for V0.)
+        (\v -> (read . Imports.tail . show $ v, v)) <$> [minBound @Version ..]
+      explicit =
```

Comment:
I don't like how easy it is to forget or make a mistake about this, but I don't have a better idea in mind at the moment. 😰

Reply:
what do you mean?  how would you do a mistake here that'd go uncaught by the CI?

Reply:
I mean forgetting to add / update versions and it not being caught by GHC/ type checking. But hopefully CI is enough to prevent us from messing up 🤞